### ISSUE TITLE
Lambda-expression closure includes ghost variables

### ISSUE THREAD

**Issue by RustanLeino** [dafny-lang/dafny#2872]
Consider the following program fragment:

``` dafny
module {:options "/functionSyntax:4"} Function4Syntax {

  method Main() {
    print F(2, 5, 9)(), "\n"; // 11
  }

  function F(a: int, ghost b: int, c: int): () -> int
  {
    () => G(a, b, c)
  }

  function G(a: int, ghost b: int, c: int): int {
    a + c
  }
}
```

The lambda expression `() => G(a, b, c)` in function `F` uses ghost parameter `b`. Alas, this parameter is included in the target code (for all target compilers). The malformed target code causes a message like:

```
> dafny test.dfy

Dafny program verifier finished with 1 verified, 0 errors
Errors compiling program into test
(1846,15): error CS0103: The name 'b' does not exist in the current context
```

**Comment by cpitclaudel** [comment]
Potential duplicate of https://github.com/dafny-lang/dafny/issues/2690 and https://github.com/dafny-lang/dafny/issues/2852 ?

**Comment by RustanLeino** [comment]
Yep, sorry, it's a duplicate and it's been fixed.
Fixed by #2854.
